specification/refarch.adoc

```diff
@@ -147,311 +189,315 @@ assigned to the TVM by the VMM.
+
+When a TVM is under self-hosted debugging - on a transition to TVM
+execution, the TSM-driver must set up the trigger CSRs for the TVM. For TVM
+debugging, the TSM-driver may inhibit M and S/HS modes in the triggers. On
+transitions back to the OS/VMM, the TSM-driver will save the trigger CSRs
+and associated debug states, thus not leaking any information to non-TEE
+workloads. TVM self-hosted debug may be enabled from TVM creation time or
+may be explicitly opted-into during execution of the TVM. The TSM may
+invoke the TSM-driver to set up a TVM-specific trigger CSR state (per the
 configuration of the TVM).
 
 *Performance Monitoring considerations*
 
-By default the TSM and all TVMs run with performance monitoring suppressed. 
-If a TVM runs in this default mode (opted out of performance monitoring), 
-on a transition to the TVM, the TSM-driver enforces this via inhibiting the 
-counters (using mcountinhibit). 
+By default the TSM and all TVMs run with performance monitoring suppressed.
+If a TVM runs in this default mode (opted out of performance monitoring),
+on a transition to the TVM, the TSM-driver enforces this via inhibiting the
+counters (using mcountinhibit).
 
-The TVM may opt-in to use performance monitoring either at initialization 
-or post-init.
+The TVM may opt-in to use performance monitoring either at initialization or
+post-initialization of the TVM.
 
-If the TVM has opted-in to performance monitoring, the TSM may invoke the 
+If the TVM has opted-in to performance monitoring, the TSM may invoke the
 SBI PMU extension (via TSM-driver) or Supervisor counter delegation extension to
-establish a TVM-specific performance monitoring controls (counters, event selectors). 
-However, the TVM must use SBI PMU extension unless TSM supports full trap & emulate
-support for the hpmcounter related ISA extensions. The TSM will assign a virtual
-counter to the TVM for the events requested to be monitored by the TVM in either
-approach. The TSM needs to manage a mapping between the virtual and physical counters
-as well. It must not delegate the LCOFI interrupt (via hideleg[13]=1) for the TVM
-and use the interrupt filtering mechanism defined in the Advanced Interrupt
-Architecture (AIA) to inject the LCOFI interrupt when the physical counter corresponding
-to the virtual counter overflows. The physical counters naturally inhibit counting 
-in S/HS and M. The TSM must save and  clear counter/event selector values as control
-transitions to the VMM or a different TVM that is using hpm. On a transition back to
-the host OS/VMM, the TSM must restore the saved hardware performance monitoring event 
-triggers and counter enables. If the TSM uses the SBI PMU extension instead of Supervisor
-counter delegation, the TSM-driver needs to perform the save/restore on behalf of the TSM. 
+establish a TVM-specific performance monitoring controls (counters, event
+selectors). However, the TVM must use SBI PMU extension unless TSM supports full
```

Comment:
Do we need specify this "counters/event selectors" ? We need more state to be managed. How about  
"establish a TVM-specific state that allows performance monitoring in a TVM" or something similar

Reply:
added